### PR TITLE
CONTRIBUTING: describe merge strategy

### PR DESCRIPTION
**Pull request by msanft**
This adds some instructions for maintainers on how PR's should be handled. They should pass CI checks and be merged through a rebase, generally. There might be more to add here in the future.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -103,3 +103,10 @@ just --list
     ```sh
     nix fmt
     ```
+
+## Handling Pull Requests as a Maintainer
+
+Pull requests should meet all check targets before they can be merged.
+When merging, please merge through a rebase of the commits of the PR onto main. GitHub calls
+this option `Rebase and merge`. The only exception to this are dependency bumps, which may
+be merged by squashing their commits.
```
Comment (burgerdev) [inline CONTRIBUTING.md:112]: If we codify this, we also need to describe the commit format and grouping.

However, I'm not in favour of this, as this results in additional back-and-forth between contributor and maintainer, where a squash with an appropriate message would have been sufficient.

Comment (msanft) [reply CONTRIBUTING.md:112]: I'm much in favor of squash-merging too, but I think think there's dissent around this. Should contributions of external commiters be an exception (i.e. squash-merged) too? What do you think, @katexochen?

Comment (katexochen) [reply CONTRIBUTING.md:112]: From my perspective, there is no need to formally specify such minor things. Edgeless devs can look at the current practices and adapt to lived practice. Reviewers can request changes if something doesn't match our usual workflow.
For external contributions, this shouldn't apply anyway.

**Comment by katexochen** [comment]
closing as not planned

**Pull request closed without merging**